omnigibson/prims/xform_prim.py

```diff
@@ -187,12 +187,22 @@ def set_position_orientation(self, position=None, orientation=None):
             parent_scale_tf = np.eye(4)
             parent_scale_tf[:3, :3] = np.diag(parent_scale)
 
+            # Combine translation, rotation, and scale to get the parent world transform
+            # Parent World Transform Matrix:
+            # | R_parent * S_parent   T_parent |
+            # |          0            1        |
             parent_world_transform = T.pose2mat(parent_pose) @ parent_scale_tf
         # When flatcache is off, we can simply use USD to get the parent's world transform
         else:
             parent_world_tf = lazy.pxr.UsdGeom.Xformable(lazy.omni.isaac.core.utils.prims.get_prim_parent(self._prim)).ComputeLocalToWorldTransform(lazy.pxr.Usd.TimeCode.Default())
             parent_world_transform = np.transpose(parent_world_tf)
 
+        # Calculate the local transform from parent to current prim
+        # Inverse of parent world transform multiplied by the current world transform
+        # Note: only the parent's scale matters for the local transform
```

Comment:
This is great! maybe explicitly specify here that Pose(X) = Translation(X) * Rotation(X) * Scale(X) so it's clear how it's decomposed